app_feup/lib/view/widgets/ScheduleEventRectangle.dart

```diff
@@ -24,16 +26,38 @@ class ScheduleEventRectangle extends StatelessWidget{
         child: Column(
           mainAxisSize: MainAxisSize.max,
           children: <Widget>[
-            new Container(
-              margin: EdgeInsets.only(top: 12.0),
-              child: new Text(this.subject, style: Theme.of(context).textTheme.display1.apply(fontWeightDelta: 2)),
-            ),
+            this.createTopRectangle(context),
             this.createBottomRectangle(context)
           ],
         ),
       ),
     );
   }
+
+  Widget createTopRectangle(context){
+    if (type == null){
+      return Container(
+        margin: EdgeInsets.only(top: 12.0),
```

Comment:
Can there be any lecture that has a null type?

Reply:
All lectures have a non-null type, however, the ScheduleEventRectangle is also used in the exams, which at the moment don't have a specified type. That check is used in those cases.